lib/fastly.js

```diff
@@ -20,45 +20,41 @@ program
   .description(`Purge all fastly content.`)
   .alias(`pa`)
   .action(() => {
-    co(function *() {
-      if (!program.apikey) {
-        program.apikey = yield coPrompt(`${fastlyApiKeyDescription}: `);
-      }
-      if (!program.serviceid) {
-        program.serviceid = yield coPrompt(`${fastlyServiceIdDescription}: `);
-      }
-      const fastly = require(`fastly`)(program.apikey);
-
-      if (program.hardpurge) {
-        fastly.purgeAll(program.serviceid, handleFastlyResponse(`All content purged.`));
-      }
-      else {
-        fastly.softPurgeAll(program.serviceid, handleFastlyResponse(`All content purged.`));
-      }
-    });
+    if (!fastlyApiKeyPresent(program)) {
+      return;
+    }
+    if (!fastlyServiceIdPresent(program)) {
```

Comment:
I was so focused on the instances where I only had to check for the key, I kept them separate, but yes, I will combine them where both are needed.